appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt

```diff
@@ -296,19 +299,43 @@ class LoggedInFlowNode @AssistedInject constructor(
             is NavTarget.Room -> {
                 val callback = object : JoinedRoomLoadedFlowNode.Callback {
                     override fun onOpenRoom(roomId: RoomId) {
-                        backstack.push(NavTarget.Room(roomId))
+                        backstack.push(NavTarget.Room(roomId.toRoomIdOrAlias()))
                     }
 
                     override fun onForwardedToSingleRoom(roomId: RoomId) {
                         coroutineScope.launch { attachRoom(roomId) }
                     }
 
+                    override fun onPermalinkClicked(data: PermalinkData) {
+                        coroutineScope.launch {
```

Comment:
If nothing is suspending I'd remove the launch

Reply:
Done